### PR TITLE
library: Compute Rust exception class from its string repr

### PR DESCRIPTION
**Pull request by workingjubilee**
Noticed this awkwardness while scanning through the code. I think we can do better than that.

### CODE REVIEW

**Review by Noratrieb — APPROVED** [review]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cb887fcb649031551e39c2aef746b6a4

File: library/panic_unwind/src/gcc.rs [hunk 1]
```diff
@@ -107,7 +107,4 @@ pub unsafe fn cleanup(ptr: *mut u8) -> Box<dyn Any + Send> {
 
 // Rust's exception class identifier.  This is used by personality routines to
 // determine whether the exception was thrown by their own runtime.
-fn rust_exception_class() -> uw::_Unwind_Exception_Class {
-    // M O Z \0  R U S T -- vendor, language
-    0x4d4f5a_00_52555354
-}
+const RUST_EXCEPTION_CLASS: uw::_Unwind_Exception_Class = u64::from_be_bytes(*b"MOZ\0RUST");
```
Comment (bjorn3) [inline library/panic_unwind/src/gcc.rs:110]: Pre-existing, but should we really reverse the stored order between little-endian and big-endian systems? We currently show `MOZ\0RUST` on big-endian systems and `TSUR\0ZOM` on little-endian systems when interpreted as string/doing a hexdump.

Comment (workingjubilee) [reply library/panic_unwind/src/gcc.rs:110]: I noticed that when I poked at it. :eyes: It is funny to me. All the other runtimes make the same mistake, so I initially decided not to change it. I think I wanted to actually look at the data through a hex dump before doing so.

Comment (Xiphoseer) [reply library/panic_unwind/src/gcc.rs:110]: Maybe a relevant datapoint: `windows-drivers-rs`' `wdk-alloc` specifically chose `u32::from_ne_bytes(*b"rust")` for their allocator tag: https://github.com/microsoft/windows-drivers-rs/blob/c3b7c4aa06e43e928f27eb704f76932688802921/crates/wdk-alloc/src/lib.rs#L50

Comment (RalfJung) [reply library/panic_unwind/src/gcc.rs:110]: Maybe it's not a mistake, maybe the type of this should be thought of as `[u8; 8]` rather than `u64`.

Comment (bjorn3) [reply library/panic_unwind/src/gcc.rs:110]: The current impl is a mistake if it should be interpreted as `[u8; 8]` as currently little endian systems put the string in reverse order, while big endian systems put it in the correct order.

Comment (RalfJung) [reply library/panic_unwind/src/gcc.rs:110]: Oh right, that would have to use `from_ne_bytes`... then transmuting back to a `u8` array would always give the same results.

Comment (Noratrieb) [reply library/panic_unwind/src/gcc.rs:110]: i assume it was just someone filling out the u64 literal in left to right order, as the previous comment described.

Comment (workingjubilee) [reply library/panic_unwind/src/gcc.rs:110]: cool, I checked the hexdump emitted and yeah, it is indeed backwards. silly Rust.

**Comment by nbdd0121** [comment]
Note that the Itanium exception ABI says that the convention is that high bit are for vendor and low bits are for language: https://itanium-cxx-abi.github.io/cxx-abi/abi-eh.html.

Perhaps put a relnote to say that the langauge is now renamed to MOZ? ;)

**Pull request merged**